### ISSUE TITLE
IKEA Trådfri motion sensor detection

### ISSUE THREAD

**Issue by kabal00** [dresden-elektronik/deconz-rest-plugin#1676]
Hi all,

I have a strange behavior of my IKEA Trådfri motion sensor which has been perfectly synchronized with the deConz software (installed on raspberry pi 3B+). But after a motion has been detected all of my bulbs within deconz will be shown as triggered (ON), but they are definitely OFF. The bulbs are from OSRAM and IKEA and I am using Homebridge which works great, except this issue.

Does anybody know what I need to do? 

Thanks in advance

**Comment by Cybersmurfen** [comment]
I got the same behaviour. All lights shows on when a motion is detected. 

It is the new waterproof detector - FW: 2.0.022 - the old detectors runs FW: 1.2.214

**Comment by Cybersmurfen** [comment]
Well. I deleted the Trådfri sensor from deconz. If I don't reset the sensor it will keep triggering all lights.

**Comment by avakarev** [comment]
I have issues with new model of motion control too.
The old model E1525 (2017) is fine
The new model E1745 (2019) cause deconz to report bulbs are on while they are off. Also deconz doesn't report motion events following the first "presense: true" till "presense: false".

**Comment by ebaauw** [comment]
Maybe the new model sends different ZigBee commands/command parameters?  It’s definitely different hardware, no dials, no duration setting, but a 30% vs 100% brightness setting.  Do you have the sensor on Day or on Night setting?

**Comment by avakarev** [comment]
@ebaauw Hard to say if it's on Day or on Night, there is no indicator anymore on the back. Now there is just a toggliing button, and I think it's supposed to be indicated by gateway/app what's the current settings. In my case, it's set by default, I didn't modify it. The older E1525 has default mode is set to "on Day". Here is how back side looks of E1745 now: 
<img width="241" alt="e1745" src="https://user-images.githubusercontent.com/745407/61202190-2599bf80-a6e7-11e9-975e-c53927356f3c.png">

**Comment by avakarev** [comment]
@ebaauw This is how response of `GET /sensors/<id>` looks:
```js
{ config:
      { alert: 'none',
        battery: 87,
        delay: 180,
        duration: 60,
        on: true,
        reachable: true },
     ep: 1,
     etag: '32101bc3e3b2992cb8f7efd5e68cb04d',
     manufacturername: 'IKEA of Sweden',
     modelid: 'TRADFRI motion sensor',
     name: 'Bathroom Motion Sensor R',
     state:
      { dark: true,
        lastupdated: '2019-07-14T21:08:45',
        presence: false },
     swversion: '2.0.022',
     type: 'ZHAPresence',
     uniqueid: '14:b4:57:ff:fe:52:3c:3a-01-0006' }
```

**Comment by Cybersmurfen** [comment]
@avakarev - if you push the toggle button once, the sensor shows if it is day or night with a small red indicator just next to the button. If you push it again it changes mode.

**Comment by ebaauw** [comment]
> Hard to say if it's on Day or on Night, there is no indicator anymore on the back. Now there is just a toggliing button

The IKEA "manual" seems to suggest there's a LED that lights up.  Maybe I'm too optimistic reading it.

> This is how response of GET /sensors/<id> looks like:

Thanks.  Some observations:
- Unsurprisingly, it has the same `manufacturer name` and `model` (or the REST API plugin wouldn't have exposed it already).
- `config.delay` is consistent with the fixed duration of 3 minutes from the manual.
- What really worries me is the lack of a `config.group`, suggesting the sensor hasn't been bound to one.  The old model would select a random group on reset and send groupcast commands by default, but apparently this model sends a broadcast by default (which causes deCONZ to mark all lights as on).  But somehow the lights themselves don't turn on...
- `state.dark` is `true` which means that either:
  - The sensor is on Day mode;
  - The sensor is on Night mode, and it's actually dark enough to turn on the lights.

Can you try the following:
- In the GUI, bind the client _OnOff_ cluster of the motion sensor to a group.  Trigger the motion sensor and check that `config.group` is created.
- Run deCONZ with `--dbg-aps=2` and observe the logfile for something like:
  ```
  Jul 15 12:09:00 pi1 deCONZ[1121]: 12:09:00:346 APS-DATA.indication srcAddr: 0xe9b9, srcEp: 0x01 dstAddrMode: 1, profile: 0x0104, cluster: 0x0006, lqi: 255, rssi: -55
  Jul 15 12:09:00 pi1 deCONZ[1121]: 12:09:00:346 	asdu: 0162420058020000
  Jul 15 12:09:00 pi1 deCONZ[1121]: 12:09:00:347 button 1002 On with timed off
  ```
  The `asdu` means the following:
  ```
  0162420058020000
      | | |   + 0x0000: _Off Wait Time_: 0.0s
      | | + 0x0258: _Timed Off_: 60.0s (i.e. `state.delay` ==> 60)
      | + 00: _Accept only when on_ is not set (i.e. `state.dark` ==> `true`)
      + command: 0x42: _On With Timed Off_
  ```
- Add a light to the group, and see if the motion sensor will control it.  Even if the light won't turn on, the REST API plugin should only mark the light(s) in the group as on, instead of all lights.

**Comment by tubalainen** [comment]
> > Hard to say if it's on Day or on Night, there is no indicator anymore on the back. Now there is just a toggliing button
> 
> The IKEA "manual" seems to suggest there's a LED that lights up. Maybe I'm too optimistic reading it.
> 
> > This is how response of GET /sensors/ looks like:
> 
> Thanks. Some observations:
> 
> * Unsurprisingly, it has the same `manufacturer name` and `model` (or the REST API plugin wouldn't have exposed it already).
> * `config.delay` is consistent with the fixed duration of 3 minutes from the manual.
> * What really worries me is the lack of a `config.group`, suggesting the sensor hasn't been bound to one.  The old model would select a random group on reset and send groupcast commands by default, but apparently this model sends a broadcast by default (which causes deCONZ to mark all lights as on).  But somehow the lights themselves don't turn on...
> * `state.dark` is `true` which means that either:
>   
>   * The sensor is on Day mode;
>   * The sensor is on Night mode, and it's actually dark enough to turn on the lights.
> 
> Can you try the following:
> 
> * In the GUI, bind the client _OnOff_ cluster of the motion sensor to a group.  Trigger the motion sensor and check that `config.group` is created.
> * Run deCONZ with `--dbg-aps=2` and observe the logfile for something like:
>   ```
>   Jul 15 12:09:00 pi1 deCONZ[1121]: 12:09:00:346 APS-DATA.indication srcAddr: 0xe9b9, srcEp: 0x01 dstAddrMode: 1, profile: 0x0104, cluster: 0x0006, lqi: 255, rssi: -55
>   Jul 15 12:09:00 pi1 deCONZ[1121]: 12:09:00:346 	asdu: 0162420058020000
>   Jul 15 12:09:00 pi1 deCONZ[1121]: 12:09:00:347 button 1002 On with timed off
>   ```
>   
>   
>   The `asdu` means the following:
>   ```
>   0162420058020000
>       | | |   + 0x0000: _Off Wait Time_: 0.0s
>       | | + 0x0258: _Timed Off_: 60.0s (i.e. `state.delay` ==> 60)
>       | + 00: _Accept only when on_ is not set (i.e. `state.dark` ==> `true`)
>       + command: 0x42: _On With Timed Off_
>   ```
> * Add a light to the group, and see if the motion sensor will control it.  Even if the light won't turn on, the REST API plugin should only mark the light(s) in the group as on, instead of all lights.

Hi! Can confirm that the new IKEA sensor is behaving oddly. It does not create the config.group and "ghosts" all lights in deconz to "ON" on motion detected. Ive tried the @ebaauw way described above with no success.

How ever, can confirm that the sensor is VASTLY more sensitive than the old one. Hoping for a solution. <3

**Comment by avakarev** [comment]
@ebaauw 
> The IKEA "manual" seems to suggest there's a LED that lights up. Maybe I'm too optimistic reading it.
I can confirm the back side shows the state and it was on Day mode.

Unfortunately I don't have the access to conbee stick / deconz at the moment so can't provide the logfile / debug output. Hopefully some users of E1745 sensors can help with that.

@ebaauw Thank you for your involvement and attempt to help!

**Comment by Ergerbolt** [comment]
Hi there, 

same problem for me. Exactly same behavior.

**Comment by JustAnotherUserName87** [comment]
Hi,

I have the same problem. Unfortunately I only have the newer sensor.

**Comment by LJSven** [comment]
Same here!

**Comment by Michaelnorge** [comment]
same here!

**Comment by ebaauw** [comment]
Made my semi-annual trip to IKEA this weekend, because I needed some new batteries and I long wanted to try the outlet with on/off switch.  Unfortunately, the motion sensors were sold out completely, both old and new model. So I got a SYMFONISK instead...

Still waiting on the FYRTUR, so I might actually visit them again by the end of the month.

**Comment by Michaelnorge** [comment]
I have arround 20 Trådfri plug sockets.
I synchronized two of the new motion sensors from Trådfri and got the same problems like described here.

Even after deleting the motion sensors the items keep turning on my lights.
I needed to reset them.

After that almost all of my plug sockets were offline. I needed to resynch them.

I don´t think that there is someone who will fix this problem. 
The first entry here is about a month ago and still nothing happend...
Bummer! :-(

**Comment by ebaauw** [comment]
> Even after deleting the motion sensors the items keep turning on my lights.

That is to be expected.  The motion sensor sends a broadcast command to turn on all lights.  This works without any interference from deCONZ, so removing the motion sensor from deCONZ won't change this.  You need to remove the motion sensor from the network (by resetting the motion sensor).

> I needed to reset them.
> After that almost all of my plug sockets were offline. I needed to resynch them.

Again, this is to be expected.  When you reset the sockets, they are removed from the network.

> I don´t think that there is someone who will fix this problem.
> The first entry here is about a month ago and still nothing happend...

Becasue no-one has provided the info we need to fix it.  To gather the info myself, I need the device...

**Comment by tubalainen** [comment]
> > Even after deleting the motion sensors the items keep turning on my lights.
> 
> That is to be expected. The motion sensor sends a broadcast command to turn on all lights. This works without any interference from deCONZ, so removing the motion sensor from deCONZ won't change this. You need to remove the motion sensor from the network (by resetting the motion sensor).
> 
> > I needed to reset them.
> > After that almost all of my plug sockets were offline. I needed to resynch them.
> 
> Again, this is to be expected. When you reset the sockets, they are removed from the network.
> 
> > I don´t think that there is someone who will fix this problem.
> > The first entry here is about a month ago and still nothing happend...
> 
> Becasue no-one has provided the info we need to fix it. To gather the info myself, I need the device...

I can provide the info needed (I hope...), just tell me what you need. Screendumps of info, debug?

//T

**Comment by ebaauw** [comment]
See https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1676#issuecomment-511347951

**Comment by triumvirrr** [comment]
@ebaauw 
I tried the suggested steps.

1) I am sorry, but I did not entirely understand the first step, what to do exactly and where to look for config.group. Probably I need the GUI for this? My Raspberry runs headless at the moment.

2) My sensor sends the folllowing command when motion is detected:
09:20:11:389 APS-DATA.indication srcAddr: 0x98be, srcEp: 0x01 dstAddrMode: 1, profile: 0x0104, cluster: 0x0006, lqi: 103, rssi: -78
09:20:11:389 	asdu: 011a420008070000

3) Binding the sensor to a group of lights does not change its behaviour, still all lights are ghost-on after a movement.

I will happily make further tests, especially test 1, with your guidance.

**Comment by ebaauw** [comment]
Ad 2.  I assume the sensor is on Day mode, and it’s light enough?  This is the expected _On with timed off_ command, with _Accept only when On_ and a 3-minute timed off.  Same as the old motion sensor, I don’t understand why the lights don’t respond.  Must be something special with the group.  I’m not sure if that can be seen in the log, I probably need ZigBee sniffing output.

How did you bind the sensor for 3 if you don’t understand 1?  Yes, you need the GUI for this.

**Comment by triumvirrr** [comment]
Yes, the example was in day mode with sufficient light.

Coming back to your test #1:
I do not understand what you mean with "the client OnOff cluster of the motion sensor" and I do not know where to look for config.group. Could you assist me with some hints?
Test  #3 could be done via the web interface.

Just to be clear:
The issue is not, that the lights do not go on.
The issue is, that ALL the lights, regardless of their configuration regarding the motion sensor (in my case mostly a bunch of Osram Smart+ Plugs) are marked as on, although they (probably) are not. In fact, they keep the physical state as before but are shown to be on for 1 minute. This false state is then also passed on to Openhab which causes quite some mess.

I have collected further logs, perhaps they are useful:
Night mode:
16:48:41:728 APS-DATA.indication srcAddr: 0x98be, srcEp: 0x01 dstAddrMode: 1, profile: 0x0104, cluster: 0x0006, lqi: 247, rssi: -60
16:48:41:728 	asdu: 0130420108070000
...
16:49:49:015 aps request id: 214 prf: 0x0000 cl: 0x0031 timeout (confirmed: 1) to 0x000D6FFFFE45ADBA (0x98BE)
16:49:49:094 aps request id: 214 finished, erase from queue

Night mode, light setting "dimmed" (instead of 100%) in the motion sensor:
16:55:04:344 APS-DATA.indication srcAddr: 0x98be, srcEp: 0x01 dstAddrMode: 1, profile: 0x0104, cluster: 0x0006, lqi: 111, rssi: -77
16:55:04:344 	asdu: 0132420108070000
...
16:56:13:015 aps request id: 228 prf: 0x0000 cl: 0x0031 timeout (confirmed: 1) to 0x000D6FFFFE45ADBA (0x98BE)
16:56:13:095 aps request id: 228 finished, erase from queue

Additional information:
The sensor is shown with "power source: main" and has lost 13% of its battery in one day. I do not know if this is helpful or if it correlates to the problem.

**Comment by ebaauw** [comment]
> I do not understand what you mean with "the client OnOff cluster of the motion sensor" and I do not know where to look for config.group. Could you assist me with some hints?

In the deCONZ GUI, you see the device's end points and clusters: blue for the server clusters and grey for the client clusters.  You need to bind the client _On/Off_ cluster 0x0006 to a group, using the _Bind Dropbox_ panel.  This is what it would look like for the old model:
![Screenshot 2019-08-12 at 22 03](https://user-images.githubusercontent.com/22179355/62894515-1db45600-bd4d-11e9-88e2-34998e22f09f.png).

Make sure to wave in front of the sensor to wake it, and press _Bind_.  Wave again after half a minute or so to make the sensor (hopefully) send a command to the bound group, and the REST API should show `config.group` when GETting the `/sensors` resource.  For the old model, it would be something like:
```
$ ph get /sensors/9
{
  "config": {
    "alert": "none",
    "battery": 34,
    "delay": 60,
    "duration": 60,
    "group": "3797",
    "on": true,
    "reachable": true
  },
  "ep": 1,
  "etag": "45ba72f1e5fa02ab0e4cb28e7f83dff7",
  "manufacturername": "IKEA of Sweden",
  "modelid": "TRADFRI motion sensor",
  "name": "TRADFRI motion sensor ",
  "state": {
    "dark": true,
    "lastupdated": "2019-08-12T18:25:29",
    "presence": false
  },
  "swversion": "1.2.214",
  "type": "ZHAPresence",
  "uniqueid": "00:0b:57:ff:fe:22:cf:5f-01-0006"
}
```
> The issue is not, that the lights do not go on.
The issue is, that ALL the lights, regardless of their configuration regarding the motion sensor (in my case mostly a bunch of Osram Smart+ Plugs) are marked as on, although they (probably) are not.

The issue is that the motion sensor sends a command that deCONZ thinks turns on all lights, but that the lights seem to ignore.  This first part makes me thing it's a kind of broadcast command, which we hope to change to a groupcast by binding the sensor to a group.  The second part is really interesting, but might not be an issue if don't need the sensor to control the lights directly, without using the gateway.

> In fact, they keep the physical state as before but are shown to be on for 1 minute

I suppose until next polled by deCONZ, so the timing might differ per light and will depend on the number of lights in your network.

> I have collected further logs, perhaps they are useful:

Cool.  Yes they are.  I think I managed to confuse myself once again.  ASDU xxxx 42 00 0807 0000 is turning the lights on unconditionally (i.e. when the sensor in on _Day_, or when the sensor is on _Night_ and it's dark); ASDU xxxx 42 01 0807 0000 is prolonging the on duration when the lights were already on, or does nothing when the lights were off (i.e. sensor is on _Night_, but it's not dark).

The really interesting part is that the same payload seems to be used for the _Dimmed_ setting.  Either the sensor sends another command after this one (I'd expect cluster 0x0008 or maybe 0x0005), or it uses another broadcast address (not visible in the log).  I'm afraid this might be another IKEA special, similar to the way the remote handles changing colour or colour temperature.

The timeouts on the ZDP (profile 0x0000) requests are unrelated.  This is deCONZ trying (and failing) to get some network info from the sensor.

**Comment by ebaauw** [comment]
Oh, and it would help to have the GUI screenshots of the new model.  In particular: is it now an HA / Zigbee 3 device, like the On/Off switch, or still a ZLL device; and does it have an 0xFC7C cluster.

**Comment by BeamMeUpTo** [comment]
Hi all of you ;),

I will try to send additional information - so here first is the screenshot of the new ikea motion sensor from the deCONZ gui:

![IkeaTradfri-Motion-Screenshot_20190813_090943](https://user-images.githubusercontent.com/45270879/62922975-fe4d1580-bdac-11e9-9dd8-3e884da38f4f.png)

The Binding Dropbox is empty!

I would like to provide additional info about
`ph get /sensors/9`

but I have no clue what "ph" is! Please enlighten me!

Beam

**Comment by ebaauw** [comment]
> here first is the screenshot of the new ikea motion sensor from the deCONZ gui:

Thanks.  Some observations:
- This is yet another mac address prefix, which hasn't been whitelisted in the REST API code.  Do you actually see the sensor in the API and/or Phoscon?
- It looks a lot like the on/off switch, including 0xFC7C IKEA cluster.  I can only assume the Trådfri hub interacts with the cluster to configure the sensor;
-  0x0008 _Level Control_ client cluster.  This strengthens my suspicion that the sensor might issue _Level Control_ commands when _Dimmed_ is set.

> The Binding Dropbox is empty!

Yes, no need to drag the 0x0008 cluster to the _Source_ drop box and fill in the group ID manually.

> I have no clue what "ph" is

Just a liitle tool I wrote to interact with the Hue/deCONZ API from the command line.  It comes bundled with homebridge-hue.  You can use `curl` instead, but then you have to specify the full URI (incl. host and username/apikey) and format the JSON yourself.

**Comment by triumvirrr** [comment]
Thanks, Eric.

Binding the sensor via GUI binding dropbox to an (in my case) empty group did the trick. No Ghost-Ons anymore. I also tested with a group with lights in it and they are switched as expected. And only the lights in the group are switched.

Side effect: This group is then not visible via Web Interface anymore...

The sensor is normally visible within all GUIs and the REST API.

Here is the sensor output via REST after binding it to group 6:
```
{
"config": {
"alert": "none",
"battery": null,
"delay": 180,
"duration": 60,
"group": "6",
"on": true,
"reachable": true
},
"ep": 1,
"etag": "26704aff65b268ca9d2b4aa67fc75963",
"manufacturername": "IKEA of Sweden",
"modelid": "TRADFRI motion sensor",
"name": "TRÅDFRI Bewegungsmelder",
"state": {
"dark": true,
"lastupdated": "2019-08-13T10:06:58",
"presence": true
},
"swversion": "2.0.022",
"type": "ZHAPresence",
"uniqueid": "00:0d:6f:ff:fe:45:ad:ba-01-0006"
}
```

**Comment by BeamMeUpTo** [comment]
Yes,

my sensor Sz.Motion is visible in phoscon!

After the tip with LevelControl Cluster I changed the GroupID to 0x0006 which in my case does not exist ;) and seems to be a workaround for me.

Reason:
I noticed in FHEM that everytime the motion sensor detects a motion HUEGroup0 got triggered in Event Monitor of FHEM. Which results in pseudo on-switching of Osram Smart+ Plugs. (belonging to HUEGroup0!)

This effect is now gone. In fhem I can use the IKEA Motion Sensor without any pseudo switching!

Beam

**Comment by LJSven** [comment]
I have the same behaivor - what do i have to do to fix it? I´m using ioBroker, Phoscon and deconz

**Comment by BeamMeUpTo** [comment]
Workaround:

Connect IKEA Motion Sensor to phoscon app as normal.

Start deCONZ GUI.

Choose sensor and activate the right dot! Register panel opens.

Activate the Binding Dropbox Window: Menu --> Panels --> Bind Dropbox

Now drag "0008 LevelControl" to Source in the Bind Dropbox Window.

Change the GroupID to a not existing one try 0x0010!

Press Bind!

![IkeaTradfri-Motion-Gruppe-setzen](https://user-images.githubusercontent.com/45270879/62936141-4ed26c00-bdc9-11e9-962e-746607f10926.png)

**Comment by ebaauw** [comment]
Cool!  So we need the REST API plugin to pick a random unused group and create the binding upon pairing the sensor.

> Now drag "0008 LevelControl" to Source in the Bind Dropbox Window.

That's actually a lucky typo on my part - I meant the 0x0006 (_On/Off_) cluster.  Although I've seen other devices that use the binding for one cluster for the other clusters as well.

> Press Bind!

Make sure the sensor is awake, or you might get a timeout.

> my sensor Sz.Motion is visible in phoscon!

@BeamMeUpTo, I'm happy for you, but I don't understand how this is possible.  Can you please double-check that the `uniqueid` indeed starts with `14:b4:57` (and not with `00:0d:6f` like @triumvirrr 's)?

**Comment by LJSven** [comment]
I don´t have the GUI installed. Is there any other way?

**Comment by BeamMeUpTo** [comment]
@ebaauw 
So, I double-checked ;)

I bought two of those sensors last weekend, and both of them have the 14:b4:57. and the known problem of pseudo-switching.

![IkeaTradfri-Motion-2x](https://user-images.githubusercontent.com/45270879/62941360-1b4a0e80-bdd6-11e9-976f-bcbecbe86abf.png)

I tried to drag "0006 on/off" to source and afterwards checked the GroupID for the other cluster - it is everywhere GroupID 0x0006

@LJSven:
I don't know if there is another way to change GroupID - Eric?

But I had to install the deconzGUI on my raspi, too and use it with vncviewer, because of headless. Not nice but it works ;)

**Comment by ebaauw** [comment]
> afterwards checked the GroupID for the other cluster - it is everywhere GroupID 0x0006

You the group ID displayed is the input for new bindings.  There's no way to retrieve (the group ID from) an existing binding in the GUI.  The only way to check is to trigger the sensor and check what the API reports for `config.group`.

> I don't know if there is another way to change GroupID - Eric?

You need to create a binding by sending the corresponding ZigBee command to the sensor or switch.  The only way to do that manually in deCONZ is through the GUI.  The REST API does this automatically for whitelisted devices, so normally you wouldn't bother.

If you want more control, you could issue the ZigBee bind command and query the binding table through the deCONZ CLI plugin, see https://github.com/ma-ca/deconz-cli-plugin.  But that's not for the faint-hearted: keep your ZigBee documentation and binary/hex - decimal converter at hand.

**Comment by ebaauw** [comment]
> I don't understand how this is possible.

Use the Source, Luke:
https://github.com/dresden-elektronik/deconz-rest-plugin/blob/254f1a8e530c635f9d430dfaddf45d9e87d997b7/de_web_plugin.cpp#L6720-L6721
So the REST API code checks for matching manufacturer code OR matching mac address.
@manup, I think this is a bug; shouldn't both match (i.e. L6720 should end in `&&` instead of `||`)?

**Comment by Ergerbolt** [comment]
Hi, i think you all have a solution for the Problem. Very nice! Thank you! If im Right, i Need dCONZ gui on my pi. How can i install this program on hassio? :)

**Comment by ebaauw** [comment]
The deCONZ GUI is already installed, as it's integral part of deCONZ.  Depending on how you start deCONZ, it shows the GUI (by default) or not (`-platform minimal` option).  If you use `systemctl` to start deCONZ as a service (normal Raspberry Pi installation), deCONZ provides a `deconz.service` and a `deconz-gui.service`.  Make sure to enable/start only one of these and disable the other.

Of course you need to start the graphical environment of your OS to run the GUI.  No clue how to do that on hassio.

**Comment by LJSven** [comment]
Can i do the changes on another PI or do i have to do it on the same where phoscon is running?

**Comment by ebaauw** [comment]
Phoscon is a web app that runs in your browser.  In connects to deCONZ running on your Pi over the REST API.  The GUI runs on that same Pi, as it's integral part of deCONZ.

You can run deCONZ with the GUI enabled on a headless Raspberry Pi, using VNC to access the graphical environment remotely.  I actually use macOS screen sharing to interact with my Raspberry Pi-s, instead of the VNC client.  And Mocha VNC on my iPad.

**Comment by triumvirrr** [comment]
Can I sniff the command when I just have one RaspBee hardware which is used for deconz, obviously? If so, I will gladly do it - perhaps with one or two hints on which tool to use :-)

**Comment by LJSven** [comment]
When i connect to my Tinkerboard via VNC it shows the following message ->

> Xsession: unable to start X session --- no "/root/.xsession" file, no 
"/root/.Xsession" file, no session managers, no window managers, and no 
terminal emulators found; aborting.

**Comment by ebaauw** [comment]
> Can I sniff the command when I just have one RaspBee hardware which is used for deconz, obviously?

No, you need a second ZigBee radio device, typically with sniffing firmware.  I use a ConBee with ZShark, see https://github.com/dresden-elektronik/deconz-rest-plugin/issues/405, on a separate Pi.

> When i connect to my Tinkerboard via VNC it shows the following message

Looks like the graphical environment hasn't been started.  I would guess it's installed, as the VNC client seems to find a VNC server on the Tinkerboard to talk to.

No clue how to start the graphical environment on a Tinkerboard; for the Raspberry Pi is through `sudo raspi-config` (_3 Boot Options_ and then _B1 Desktop / CLI_).  Same to enable the VNC server (_5 Interfacing Options_ and then _P3 VNC_).

**Comment by Ergerbolt** [comment]
i got it. Now i bonded 0008 Cluster to Gruup 0x0010. But now my Group "Bath" is away :-D

**Comment by BeamMeUpTo** [comment]
0x0010 was an example, which was sufficent for my setup.

If you have more groups than 10 - you have to choose a higher value! ;)

**Comment by Ergerbolt** [comment]
Ok, hmm, i will try. When i use 0x0006 some lights - from group 6 - are switched on. 
Do you know how i can get my groups in the phoscon app back?

**Comment by BeamMeUpTo** [comment]
0x0006 is lower than 0x0010, so it is clear, that you switch on another group, if you had already difficulties with 0x0010 - try 0x0100!

**Comment by Ergerbolt** [comment]
Sure, you are right, i just wanted to tell, that ... hmm i don't know what i wanted to say. Wich cluster do i have to change? The "Level control" or the "on/off" cluster? I'm not sure because there are two descriptions in the earlier threads?

**Comment by BeamMeUpTo** [comment]
on/off cluster is the save way

**Comment by Cybersmurfen** [comment]
Well. I'm connected to homeassistant as well and are using the event listener to trigger automations.

I decided to add a new motion sensor and except for all above, it looks like it doesn't come out any events from deconz at all while the sensor is connected.

And yes, there is 3 buttons on the back. One toggler for day/night, one toggler for wide detection and one link button. When toggling the button, a small red light indicates which mode is selected.

The new motion sensor is identified as an old one, and the firmware is 2.0.22

In the phoscon app, you can add it to a motion control and it is working as expected, but all other lights are "ghosted on".

And to remove it, just remove it from the app and then reset the sensor.

**Comment by LJSven** [comment]
It looks like, if you switch the day & night button to "night" the problem is not happening.

**Comment by BeamMeUpTo** [comment]
I played around with REST API so I could really verify my groupID Settings: (In my case it made no difference if I write 0x0006 to the wrong Level Cluster or into the right ON/OFF Cluster)

Here is the output from both Sensors (FYI Erik):

`"6": {
    "config": {
      "alert": "none",
      "battery": 100,
      "delay": 180,
      "duration": 60,
      "group": "6",
      "on": true,
      "reachable": true
    },
    "ep": 1,
    "etag": "17ce7f393dd778ad96aa7a3d5336eba5",
    "manufacturername": "IKEA of Sweden",
    "modelid": "TRADFRI motion sensor",
    "name": "TRÅDFRI Bewegungsmelder",
    "state": {
      "dark": true,
      "lastupdated": "2019-08-14T06:10:17",
      "presence": false
    },
    "swversion": "2.0.022",
    "type": "ZHAPresence",
    "uniqueid": "14:b4:57:ff:fe:3e:b1:47-01-0006"
  },
  
    

"9": {
    "config": {
      "alert": "none",
      "battery": 100,
      "delay": 180,
      "duration": 60,
      "group": "6",
      "on": true,
      "reachable": true
    },
    "ep": 1,
    "etag": "d84a07018017428aea5f33be24c7827b",
    "manufacturername": "IKEA of Sweden",
    "modelid": "TRADFRI motion sensor",
    "name": "Sz.Motion",
    "state": {
      "dark": true,
      "lastupdated": "2019-08-13T20:25:46",
      "presence": false
    },
    "swversion": "2.0.022",
    "type": "ZHAPresence",
    "uniqueid": "14:b4:57:ff:fe:3e:ad:0d-01-0006"
`

**Comment by tubalainen** [comment]
Got it to work! Sorry @manup didnt understand "drag and drop to source". WIth some trial and error I got the new IKEA Motion sensor to work via the REST API to Home Assistant!

This were the steps for the deconz REST-API:
0.) You need to reach the GUI. I use Marthoc´s docker (https://github.com/marthoc/docker-deconz) and have enabled VNC server in the docker. I then use the vncviewer application windows to connect to the Deconz GUI. 
1.) Add the sensor just like normal in Phoscon
1.5) On the back of the sensor, toggle the night/day mode so that the crested moon lits up
2.) Connect using VNC (in my case) to your deconz instance
3.) Go to "Panels - Bind dropbox"
4.) Locate your sensor amongst the rest of your nodes in the GUI
5.) Push the far right of the two grey circles on your sensor note
6.) Drag and drop the "0004 Groups" cluster to SOURCE in the "Bind dropbox" panel
7.) Select "Group" and enter the value 0x0100
8,) Trig the sensor my moving (so it is awake) and push "bind"
9.) Restart deconz and home assistant and you should be all good to go.
![Capture1](https://user-images.githubusercontent.com/26795429/63045841-80cef580-bed1-11e9-80f4-168ba69ef51a.PNG)
![Capture4](https://user-images.githubusercontent.com/26795429/63045842-80cef580-bed1-11e9-82ca-019e0d11a3ae.PNG)
![Capture3](https://user-images.githubusercontent.com/26795429/63045843-80cef580-bed1-11e9-945e-e872203ef954.PNG)
![Capture2](https://user-images.githubusercontent.com/26795429/63045844-81678c00-bed1-11e9-89ea-ab69bf0e0f35.PNG)

**Comment by Cybersmurfen** [comment]
I can confirm that tubalainens solution works like a charm.

Tackar!

**Comment by ahd71** [comment]
I also confirm that the workaround with binding a new group worked for me! Thanks all.

Some clarification (0x0010 is likely hex 10 which is 16), but perhaps most improtant I added a last step of "walk in front of sensor" after pressing the bind button. It was NOT necessary to restart deconz or home assistant for me after the binding as hinted in some other guide.

- Connect IKEA Motion Sensor to phoscon app as normal.
- Start deCONZ GUI.
- Choose sensor and activate the right dot! Register panel opens.
- Activate the Binding Dropbox Window: Menu --> Panels --> Bind Dropbox
- Now drag "0006 On/Off" to Source in the Bind Dropbox Window.
- Change the GroupID to a not existing one try 0x0010! (Example!!! Only use if you have less than 0x0010 Groups!)
- Press Bind!
- Walk in front of the sensor - you should see a confirm message in the bind dialog if it works - otherwise a timeout message

**Comment by Cybersmurfen** [comment]
@adh71 - "0006 On/Off" ?
tubalainen stated "0004 Groups" and that worked for me.
Typo or did both work?

**Comment by ahd71** [comment]
I tried only 0006

Den tors 15 aug. 2019 00:33Cybersmurfen <notifications@github.com> skrev:

**Comment by ahd71** [comment]
I tried only 0006

**Comment by Ergerbolt** [comment]
I think, its both working fine, because i took 0008. BeamMeUpTo already confirmed this as well.

**Comment by manup** [comment]
> > I don't understand how this is possible.
> 
> Use the Source, Luke:
> https://github.com/dresden-elektronik/deconz-rest-plugin/blob/254f1a8e530c635f9d430dfaddf45d9e87d997b7/de_web_plugin.cpp#L6720-L6721
> 
> 
> So the REST API code checks for matching manufacturer code OR matching mac address.
> @manup, I think this is a bug; shouldn't both match (i.e. L6720 should end in `&&` instead of `||`)?

Should be fine, first branch checks if the node descriptor provides a valid manufacturer code and as alternative the mac prefix based checks are applied, therefore the `||`.

Our new IKEA sensors should arrive here soon so will do some more tests to automate the current manual configuration.

**Comment by ebaauw** [comment]
@manup, Welcome back!

I meant the `||` at the end of line 6720, not the one on line 6721.  I think the logic should be:
```
!isNull AND manufacturerCode AND (mac OR checkVendorMac)
```
instead of
```
(!isNull AND manufacturerCode) OR (mac OR checkVendorMac)
```

**Comment by martinkrulltott** [comment]
Hey, I'm experiencing this issue as well, running on a Raspberry Pi 3B+ and the deconz docker image (https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1767). It looks like you guys are accessing a GUI, but I'm running this headless in Docker, anyway to fix this through the console?

**Comment by Ergerbolt** [comment]
I think you can use the vnc device protokoll

**Comment by ebaauw** [comment]
Picked up a new Trådfri motion sensor today.  Had some challenges to get deCONZ to read the descriptors, but finally it displayed in the GUI.  Out of the box, it doesn't send broadcast commands, but groupcast commands to group 0x0000.  Seems like PR #1555 broke more than when we realised (see #1612).  The REST API still seems to think the lights are subscribed to group 0x0000 (hence the API reports all lights on when detecting motion), where, in fact, they aren't (so the lights don't actually turn on).  After joining a light to group 0x0000, it turns on alright when motion is detected.
```
Frame 1628: 128 bytes on wire (1024 bits), 128 bytes captured (1024 bits) on interface 0
Ethernet II, Src: Raspberr_dd:8a:80 (b8:27:eb:dd:8a:80), Dst: Apple_ca:3b:53 (a4:5e:60:ca:3b:53)
Internet Protocol Version 4, Src: 192.168.76.83, Dst: 192.168.76.41
User Datagram Protocol, Src Port: 40635, Dst Port: 17754
ZigBee Encapsulation Protocol, Channel: 11, Length: 54
IEEE 802.15.4 Data, Dst: 0x91e3, Src: 0xbecf
    Frame Control Field: 0x8861, Frame Type: Data, Acknowledge Request, PAN ID Compression, Destination Addressing Mode: Short/16-bit, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: Short/16-bit
    Sequence Number: 90
    Destination PAN: 0x476f
    Destination: 0x91e3
    Source: 0xbecf
    [Extended Source: 14:b4:57:ff:fe:3e:ad:28 (14:b4:57:ff:fe:3e:ad:28)]
    [Origin: 244]
    Frame Check Sequence (TI CC24xx format): FCS OK
ZigBee Network Layer Data, Dst: Broadcast, Src: 0xbecf
    Frame Control Field: 0x2248, Frame Type: Data, Discover Route: Enable, Security, End Device Initiator Data
    Destination: 0xfffd
    Source: 0xbecf
    Radius: 12
    Sequence Number: 184
    [Extended Source: 14:b4:57:ff:fe:3e:ad:28 (14:b4:57:ff:fe:3e:ad:28)]
    [Origin: 244]
    ZigBee Security Header
ZigBee Application Support Layer Data, Group: 0x0000, Src Endpt: 1
    Frame Control Field: Data (0x0c)
    Group: 0x0000
    Cluster: On/Off (0x0006)
    Profile: Home Automation (0x0104)
    Source Endpoint: 1
    Counter: 57
ZigBee Cluster Library Frame
    Frame Control Field: Cluster-specific (0x01)
    Sequence Number: 22
    Command: Unknown (0x42)
```

**Comment by ebaauw** [comment]
> Had some challenges to get deCONZ to read the descriptors

It seems to be extremely sleepy.  When triggering motion, the radio is awake only during sending the commands; immediately thereafter it becomes unresponsive again.  It does support _Poll Control_, @manup, maybe we could tweak this?
![Screenshot 2019-08-22 at 20 21](https://user-images.githubusercontent.com/22179355/63539511-928c4a80-c51a-11e9-8c88-f22ac645fdbb.png)

Also, it seems to take well over a minute to "recharge", before sending a next command.  The "read the _Basic_ cluster attributes_ trick to create the API resource might take a lot of attempts.

Next to the _On with Timed Off_ commands, it sends _Move to Level with On/Off_ commands.  Level 0x76 for 30% and level 0x254 for 100%.  It only sends these for a limited period / limited number of times (?) after changing the setting.

**Comment by manup** [comment]
> @manup, Welcome back!
> 
> I meant the `||` at the end of line 6720, not the one on line 6721. I think the logic should be:
> 
> ```
> !isNull AND manufacturerCode AND (mac OR checkVendorMac)
> ```
> 
> instead of
> 
> ```
> (!isNull AND manufacturerCode) OR (mac OR checkVendorMac)
> ```
> 
> Otherwise the whole mac address check seems redundant, since the manufacturer code already matches.

I think the OR is needed to support devices where the node descriptor and manufacturer code wasn't (or can't) be queried. I'm afraid that when the node descriptor is a requirement the function might fail for some devices.

But really hard to tell for sure, perhaps the device map should be extended with flags/attributes per modelid on which data is available for a certain device?

**Comment by manup** [comment]
>It seems to be extremely sleepy. When triggering motion, the radio is awake only during sending the commands; immediately thereafter it becomes unresponsive again. It does support Poll Control, @manup, maybe we could tweak this?

If the Poll Control cluster is implemented properly in the IKEA firmware we can setup the polling as needed (the cluster is pretty cool actually). For a reference this is already done for the Samsung sensors and works great there.

**Comment by ebaauw** [comment]
The above commit creates a group for the (new) motion sensor on pairing.

**Comment by ebaauw** [comment]
Hm, on re-pairing, the group is created and the ZHAPresence resource shows `config.group`, but it still sends commands to group 0x0000.  Looks like the binding didn't make it.

**Comment by ebaauw** [comment]
There's always one more place to whitelist a device.  Above commit fixes this.

**Comment by Mikha495** [comment]
Sorry for question, but what is necessary do do with my setup in order to apply this fix?

**Comment by ebaauw** [comment]
Compile the REST API plugin from the commit and re-pair the motion sensor.  Alternatively, wait for it to be included in a next version of deCONZ.

**Comment by Cybersmurfen** [comment]
It isn't included in the 67 version?

**Comment by Korhm** [comment]
Hello,
when will this version be available for raspbian strech on the repo ?
Latest available version is 2.05.65

Thank you.

**Comment by Korhm** [comment]
Hello,

I can confirm that the new release solve it.
But I had to install it on Raspbian via the beta repo.

**Comment by Mikha495** [comment]
> Hello,
> 
> I can confirm that the new release solve it.
> But I had to install it on Raspbian via the beta repo.

But Issue #1676 is not mentioned in the change log of release x.67.

**Comment by Cybersmurfen** [comment]
the 67 version does not include the fix.

**Comment by DieterClaeys** [comment]
I have version 02.05.67 the issue still persist.

**Comment by vandalon** [comment]
> Made my semi-annual trip to IKEA this weekend, because I needed some new batteries and I long wanted to try the outlet with on/off switch. Unfortunately, the motion sensors were sold out completely, both old and new model. So I got a SYMFONISK instead...

You know you can just order online, right? :) That's how I got my motion sensor 🗡 
I can formir I have the same issue here.

**Comment by salopette** [comment]
So for me there is still the problem that all groups go to when the sensor triggers

**Comment by DieterClaeys** [comment]
@salopette , from what I understand no fix has been released yet

**Comment by mattias1972** [comment]
Just tried 02.05.69 (in deCONZ addon v3.6 for Home Assistant) and the issues still remain it seems... Anybody that have any luck getting this to work?

**Comment by ebaauw** [comment]
Did you re-pair the sensor?

**Comment by mattias1972** [comment]
No... Will try that!!

**Comment by mattias1972** [comment]
Ok, I'll put on the dunce cap... After repairing is works!

**Comment by DieterClaeys** [comment]
So I know what to do tomorrow 😂

**Comment by simonbove** [comment]
So to confirm all problems with the new motion sensor from ikea is solved? So now we can go buy them?

**Comment by manup** [comment]
The issue is solved since version 2.05.69.
Note that the sensor needs to be re-paired.

**Comment by DieterClaeys** [comment]
Issuee still persists even after repairing the sensor.

**Comment by DieterClaeys** [comment]
Ok, seems the docker image is still on 2.05.67 - so the bugfix wont be active yet

**Comment by JensTimmerman** [comment]
I just got my conbee 2 and tradfri motion sensor (one of the new ones with digital buttons instead of analog turn buttons) 5 days ago, I am on 2.05.69. and I am unable to pair my motion sensor, 
The only way I can get it to show up in deconz is if I pair it to a lamp first, then it shows up but 'unreachable' and the lamp seems to get removed from the system aswell, I have to reset the lamp to get it to connect back to deconz but the motion sensor stays as 'unreachable'

Are there any clear steps on how to connect the motion sensor?
pressing the reset button 4 times does nothing.
Do I need to pair it to a light first and then press the button 4times?
Do I need to keep it 5cm from the conbee 2 and then press the button 4 times?
Do I need to press the button 10 seconds close to the conbee 2?

**Comment by bmelman** [comment]
> Additional information:
> The sensor is shown with "power source: main" and has lost 13% of its battery in one day. I do not know if this is helpful or if it correlates to the problem.

Same issue here: battery drained from 85 to 15% within one week.

**Comment by RestOp** [comment]
In Motion sensor i have 2,0,022 and there is no update available. Is it normal?

**Comment by gloomytree** [comment]
Hi there, 
i still have issues with the E1745 (2019, IP44). I can pair it without problems, after that it doesn't do anything. Phoscon says Status Not reachable. 
I am running a Raspberry Pi 4 with RaspBee, HASS.io and the according deconz plugin. Version is 2.05.69, so i SHOULD be fine, according to this thread, Firmware is 26050500 (but i cannot update using the Phoscon GUI).

Can someone provide some insights? Need more info?

**Comment by wucherpfennig** [comment]
I am sorry to hijack this thread but as state above hassio + deconz latest (2.05.69) works good BUT I noticed that there is a strange "cooldown" time of 60 seconds in which the sensor does not detect anything. Is this behaviour changeable?

**Comment by manup** [comment]
>I am running a Raspberry Pi 4 with RaspBee, HASS.io and the according deconz plugin. Version is 2.05.69, so i SHOULD be fine, according to this thread, Firmware is 26050500 (but i cannot update using the Phoscon GUI).

That's a very old firmware which doesn't support devices like IKEA remote and sensor well.
I'd recommend to upgrade the firmware manually to the latest version 0x26330500 as described in https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Update-deCONZ-manually

**Comment by Korhm** [comment]
> I am sorry to hijack this thread but as stated above hassio + deconz latest (2.05.69) works good BUT I noticed that there is a strange "cooldown" time of 60 seconds in which the sensor does not detect anything. Is this behaviour changeable?

I have the same behavior. Don’t know if it’s because of deCONZ or the motion sensor itself.
I’m also interrested of knowing how to avoid this « cooldown » time.

**Comment by manup** [comment]
Most likely the sensor prevents battery drain by not sending messages too often.

**Comment by gloomytree** [comment]
> 
> 
> > I am running a Raspberry Pi 4 with RaspBee, HASS.io and the according deconz plugin. Version is 2.05.69, so i SHOULD be fine, according to this thread, Firmware is 26050500 (but i cannot update using the Phoscon GUI).
> 
> That's a very old firmware which doesn't support devices like IKEA remote and sensor well.
> I'd recommend to upgrade the firmware manually to the latest version 0x26330500 as described in https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Update-deCONZ-manually

Thanks, this saved me. Altough it was kind of difficult to upgrade the firmware via Hass.io / Hass.os, it now works like a charm.

**Comment by MacJawa** [comment]
> 
> 
> Most likely the sensor prevents battery drain by not sending messages too often.

This kind of makes sense, but on the other hand, as per my experience, the sensor only waits 60 seconds after presence changed to true. Then it sends another update (after 60 seconds). Let's say the second update is presence = false. Then it does not send anything until another movement detected. As assume this behaviour based on "lastupdated" value. Therefore it probably does not send update every 60 seconds, but to be sure we need to monitor the network activity and this is above my actual skill :)
If I'm correct above, then if there is not so much movement, it will make sense to shorten the interval and still it will only send 2 updates. Or if battery drain is not a concern and one would like to have updates more often for the movement, it would be nice to have ability to set the interval to custom value.

So the question remains. Is it possible to set config.delay to custom value other then default 60?
Hopefully @ebaauw or @manup  would clarify that.

**Comment by ebaauw** [comment]
The Trådfri motion sensor doesn’t send state updates (except for battery), and it doesn’t send anything when motion is no longer detected.  It sends an _On with Timed Off_ command to the bound group.  This causes the lights in this group to turn on for the specified time.  No need for a second message, and no need to report next/continued motion before that time has almost expired.

The deCONZ REST API plugin eavesdrops on the group command and sets `state.presence` accordingly.  It clears `state.presence` autonomously after `config.duration` seconds, but this has nothing to do with the motion sensor.  Note that the API was modelled after the Hue motion sensor, which works very differently.  This sensor does send state updates, also when it no longer detects motion.

**Comment by MacJawa** [comment]
Thank you @ebaauw  for further information. Unfortunately I still don't understand how motion sensor (new one) behaves when there is continous movement for let's say 5 minutes. Let me explain. I use OpenHab because I want to build more advanced rules, not only simple grouping. At first, I set simple rules:
When presence changes to True, turn on the lights. When presence changes to False, turn off the lights. I quickly found out this wouldn't work, because even I was moving in front of sensor, lights turned off after 60 seconds. They eventually turned on again, but only after I was jumping in front of sensor for a few moments. This is not good, right? So I changed my second rule like this: When presence changes to False, wait for 60 seconds, than check presence again and act. If presence is False, turn off the lights, otherwise let it be. But this is not good as well, because when I rush into kitchen for a cup of water and I leave in 10 seconds, my lights are turned on for 2 minutes, which is kind of wasted energy and not very smart. Of course I can lower the interval and check for presence again in 10 seconds after it changes to False, hoping the sensor will send the update meanwhile, but it's still one minute of wasted energy after I left the room. 
But if that initial 60 seconds delay could be shortened to 20 seconds for example, this could solve the situation.

Back to beginning. You say that sensor only sends update when motion is detected. Does it then sleep for 60 seconds? Because according to log in openhab my presence state always changes to OFF exactly one minute after ON, regardless of the movement. Is this behaviour of the sensor or deconz API? 
And finally, is it possible to turn off lights sooner after I leave the room (as I explained above) or is it impossible with this sensor?

Thank you for any ideas.

**Comment by ebaauw** [comment]
> You say that sensor only sends update when motion is detected.

Strictly speaking, it sends a command to turn on lights, it doesn’t send a status update. But yes: only when it detects motion.

> Does it then sleep for 60 seconds?

Sleep as in shuts down its ZigBee radio: yes.  Sleep as in: won’t detect motion: I don’t know.  Also I haven’t looked in detail, how far apart it sends commands under continuous motion, but for the new sensor, it seems closer to 180 seconds than 60 seconds.

> Because according to log in openhab my presence state always changes to OFF exactly one minute after ON, regardless of the movement. Is this behaviour of the sensor or deconz API?

Again, `state.presence` is a REST API attribute that has no ZigBee equivalent for the Trådfri motion sensor.  The REST API plugin resets it to false after `config.duration` seconds.  If you set `config.duration` to 0, the REST API resets `state.presence` after `config.delay` seconds (the _OnTime_ that the sensor specified with the _On with Timed Off_ command).  Personally, I would never have exposed `config.duration` for the Trådfri sensor, since it’s highly confusing and `delay` makes more sense if you use the sensor to control lights directly.

> And finally, is it possible to turn off lights sooner after I leave the room (as I explained above) or is it impossible with this sensor?

A single motion sensor cannot detect when you leave a room.  It detects motion, not presence.  I actually use multiple motion sensor in adjacent rooms to detect leaving (motion in living room, motion in hallway, no more motion in living room => left the living room).

The granularity of the new Trådfri motion sensor is close to 3 minutes, i.e. you cannot infer no motion any sooner.

> my lights are turned on for 2 minutes, which is kind of wasted energy and not very smart.

With LED based smart lights I wouldn’t worry about that.  If you use a wired in-wall switch with tungsten or halogen lights, it’s maybe a different story - I’d replace them with (dumb) LED lights.

**Comment by oywino** [comment]
Excuse me for butting in. I tried to read this loooong thread, but somehow got lost in too many technical terms. I have two small questions:
1. Has the "problem" with ghost-on been solved by anyone?
2. I have the new model IKEA motion sensor paired with my ConBee II and integrated into HA 0.103.4
    When creating an automation, I get something called "duration" in the trigger section, with default value 00:00:00
    What is "duration", and how can I use it in an automation (what does it do and how does it work)?

**Comment by MacJawa** [comment]
Hi @oywino ,

let's think about IKEA motion sensor this way: When it detects movement, it sends to zigbee network command to turn on lights for a specific time. This specific time is the "duration" you mentioned. So if duration is set to 60 seconds (default if I'm not mistaken), then the motion sensor sends the command: hey lights, turn on for 60 seconds, then turn off.

That's it. There is no other message sent afterwards, just this one. No "turn off" message after 60 seconds or anything similar. Now deConz takes this message and set motion sensor "ON" for the duration time. Let's say it simulates the motion for a set duration. After that, when motion sensor detects movement, it sends another command to turn on lights. And this repeats.

What I did in OpenHab is this logic:
- when motion changes from False to True, I turn on lights.
- when motion changes from True to False (always after 60 seconds), I start the timer for another 30 seconds. After that time I check the motion status:
    - if status is True, I let the lights turned on and I wait for another OFF
    - if status is False, this means there was no movement for last 30 seconds and I turn off lights.

I recently build HomeAssistant besides my OpenHab instance and didn't try automation yet, but I will give it a try. I will eventually move all my autiomations to HA, so I will also check how it looks there. If you have more questions, don't hesitate to ask ;)

**Comment by ebaauw** [comment]
> This specific time is the "duration" you mentioned.

No.  The time is fixed to 3 minutes for the new model and set between 1 and 10 minutes using the dial on the old model.  This value is reported by the REST API as `config.delay`.

> What is "duration",

`config.duration` is used by the REST API plugin to reset `state.presence` to false automatically, after the given number of seconds.  This is useful for motion sensors that don’t send a message when motion is no longer detected, but highly confusing for motion sensors that do (like the Hue motion sensor) or for motion sensors that send a _On with Timed Off_ command on motion (like the Trådfri motion sensor).  For the first type, you want to configure the _PIR Occupied to Unoccupied Delay_ through `config.delay` (which is read/write for these).  For the second type you want `state.presence` to become false automatically when the lights turn themselves off, i.e. after `config.delay` (which is read-only for these).

> and how can I use it in an automation

You don’t.  Since the REST API already resets `state.presence` automatically, you’d base your automation on that.  If you want to set the duration yourself from your automation, best leave `config.duration` to 0.

> When creating an automation, I get something called "duration" in the trigger section, with default value 00:00:00

That must be an HA thing; `config.duration` takes an integer value.

Note that motion sensors typically take some time to “recharge” after detecting motion.  They won’t send another motion detected message during that period.  If you set `config.duration` too short your automation might turn the lights off while there’s still (undetected) motion, and you’re left in the dark.  This “recharge” period is hardwired in the sensor, but there is a mod for the Xiaomi motion sensor to shorten it.  For the new model Trådfri motion sensor it’s close to the three-minute delay.

**Comment by oywino** [comment]
![image](https://user-images.githubusercontent.com/18447119/71576087-a7dc3b80-2aef-11ea-924b-d736cd7cce2d.png)
This is what it looks like in HA Automation UI. Whoever programmed this dialog must have had some purpose when he/she decided to include the duration timer?

**Comment by ebaauw** [comment]
Best ask the HA people.

**Comment by oywino** [comment]
> Best ask the HA people.

I will. Thanks for taking the time to reply :)
Happy New Year

**Comment by vandalon** [comment]
I think HA triggers on motion detect and then applies own logic from that point on. So the duration is controller by HA and the stop motion trigger from the tradfri sensor is ignored.

**Comment by runevad** [comment]
> ![image](https://user-images.githubusercontent.com/18447119/71576087-a7dc3b80-2aef-11ea-924b-d736cd7cce2d.png)
> This is what it looks like in HA Automation UI. Whoever programmed this dialog must have had some purpose when he/she decided to include the duration timer?

The duration is used as a criteria for removing false positives in the trigger. So the trigger only “goes through” if motion has been detected for example x minutes straight. Perhaps not as useful when using a sensor like this, but the duration criteria is there on several things in HA. More useful for other sensors to only trigger something. For example a door that has been opened or closed for x seconds or so.

**Comment by Jagohu** [comment]
I think I figured out an easy way to solve the issue with new Tradfri motion sensor turning on all the lights. I couldn't find it anywhere else so far, so I figured I'll share it here for everyone.
1. Create a group "All lights" in DeConz (Phoscon). Put in all your lights which you DON'T want Tradfri to control automatically.
2. Select the group, then "Sensor control"
3. Add rule - name it as you like, I call it IKEA Tradfri.
4. Select "Do nothing" for every option.
![image](https://user-images.githubusercontent.com/35196588/73966479-5ad15d80-4916-11ea-866f-75d6d188d46c.png)
5. Click on this icon below the window:
![image](https://user-images.githubusercontent.com/35196588/73966517-70468780-4916-11ea-8c8e-3c8a4bcd4513.png)
6. Select IKEA Tradfri Motion sensor, to get the checkmark highlighted (as opposed to the Dimmer next to it).
![image](https://user-images.githubusercontent.com/35196588/73966562-8d7b5600-4916-11ea-8084-d7c23d25883d.png)
7. Enjoy :)

**Comment by wes1993** [comment]
Hello!! Can someone know if there is a way to change the delay from 180 to 60 for example?
Thanks a lot
Stefano

**Comment by ebaauw** [comment]
The new model Trådfri motion sensor uses a fixed delay of 180 seconds.  This is hard-coded in the sensor firmware.  The old model has a user-configurable delay, between ~60 and ~600 seconds, using the dial on the sensor.

**Comment by gloomytree** [comment]
> Hello!! Can someone know if there is a way to change the delay from 180 to 60 for example?
> Thanks a lot
> Stefano

You could try to reverse engineer the firmware.

**Comment by martinkrulltott** [comment]
_Quick update for anyone experiencing the original issue._
I had the same issue as the OP, with the lights getting set to "ON" in Home Assistant when they're actually off. Using a Raspberry Pi and deconz, with IKEA bulbs and motion detector. These two solutions https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1676#issuecomment-521437531 https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1676#issuecomment-521361772 worked the first time it happened (6 months ago), but I recently changed the batteries in one of the motion sensors which seems to have triggered the issue to come back.

This time around I tried the same solution but without success, as the binding kept failing with `failed: timeout`. However, the trick seems to be to leave the motion sensor for a while (without triggering it) and only walk past it / trigger it when you've pushed the `bind` button. My best I guess there's a throttle on how often the motion sensor sends messages so if you walk past it too soon / too often, the bind command might miss the communication slot.
With the bind `success` the issue is now resolved again 🙂

**Comment by martinkrulltott** [comment]
My second sensor died a while back and I just changed the batteries. As soon as the pairing was done my lights started to turn on and off randomly and not responding to input properly. Once again I did the method outlined above and it worked ⬆️ 👍

**Comment by oywino** [comment]
I have read tis tread and I am somewhat confused about all the issues reported - and the complexity of the advises given.
For me, using the "new" IKEA motion sensor with deConz was alsmot just as easy as using it with the IKEA HUB. And I never one had to visit deConz to get it working. I only used the Phoscon GUI.
Adding it to Phoscon was easy - just put it in pairing mode and search for it and Phoscon finds it right away.
Then I added the sensor to a room, and configured the trigger for on and off, and it immediately began to control all the lamps in the same room according to motion detection.  And with Phoscon, it was easy to change the delay from 3 to 5 minutes. 
![image](https://user-images.githubusercontent.com/18447119/102140593-c00fac80-3e5f-11eb-9821-5c0c54b0af9e.png)
Works for me!

**Comment by meros** [comment]
I just stumbled on this issue myself. I paired and configured the motion sensor without problems, but getting ghost 'ON' lights during usage. I tried binding the cluster to an empty group but when trying to do that I keep getting 'failed: table full'.

**Comment by meros** [comment]
> I just stumbled on this issue myself. I paired and configured the motion sensor without problems, but getting ghost 'ON' lights during usage. I tried binding the cluster to an empty group but when trying to do that I keep getting 'failed: table full'.

I did a reset of the device, deleted in deconz and readded. After this, the binding command worked! :)

**Comment by thecobra666** [comment]
If I'm reading this right, the motion sensor can only trigger on day OR night, not both?
